bindings/c/src/lib.rs

```diff
@@ -403,7 +403,7 @@ pub unsafe extern "C" fn opendal_operator_stat(
     let path = unsafe { std::ffi::CStr::from_ptr(path).to_str().unwrap() };
     match op.stat(path) {
         Ok(m) => opendal_result_stat {
-            meta: opendal_metadata::from_metadata(m),
+            meta: Box::into_raw(Box::new(opendal_metadata::new(m))),
```

Comment:
So the `opendal_metadata` itself contains will be a 64-byte heap-allocated. I think the self should be dropped as well.

Reply:
Hi, if I have a struct like:

```rust
#[repr(C)]
pub struct opendal_result_stat {
    pub meta: *const opendal_metadata,
    pub code: opendal_code,
}

#[repr(C)]
pub struct opendal_metadata {
    pub inner: *mut od::Metadata,
}
```

Do I need to implement `opendal_result_stat_free` to free `opendal_result_stat`?

Reply:
Is it possible for us to avoid the `Box::new()` here?

Reply:
> Is it possible for us to avoid the `Box::new()` here?

I think ideally it is possible, but not for the current scheme. Since the `opendal_meta` has to live on the heap(it outlives the current function), we are forced to use the Box::new to create this 8-byte heap pointer.

If we really want that, we have to let `meta` points at `od::Metadata` instead of `opendal_metadata`.

Reply:
> If we really want that, we have to let `meta` points at `od::Metadata` instead of `opendal_metadata`.

But I do think there may exist some magical ways to achieve this but I cannot come up with them right now :(

Maybe #[repr(transparent)]? (I used this way before but this made it hard for doxygen documentation.)

Reply:
Ok, let's figure in other PRs than.